index.js

```diff
@@ -379,24 +380,39 @@ function Argv (processArgs, cwd) {
 
   var versionOpt = null
   self.version = function (opt, msg, ver) {
-    if (arguments.length === 1) {
+    if (arguments.length === 0) {
+      ver = guessVersion()
+      opt = 'version'
+    } else if (arguments.length === 1) {
       ver = opt
-      versionOpt = 'version'
-      msg = usage.deferY18nLookup('Show version number')
+      opt = 'version'
     } else if (arguments.length === 2) {
-      versionOpt = opt
       ver = msg
-      msg = usage.deferY18nLookup('Show version number')
-    } else {
-      versionOpt = opt
     }
 
+    versionOpt = opt
+    msg = msg || usage.deferY18nLookup('Show version number')
+
     usage.version(ver || undefined)
     self.boolean(versionOpt)
     self.describe(versionOpt, msg)
     return self
   }
 
+  function guessVersion () {
+    var obj = readPkgUp.sync({
+      cwd: path.resolve(__dirname, '../')
+    })
+
+    // fallback to process.cwd() this
+    // solves the case of symlinked modules.
+    if (!obj.pkg) {
+      obj = readPkgUp.sync()
+    }
```

Comment:
there's two reasons why I think it might be worth falling back to `cwd`:

1. without handling this edge-case, we can't test the `yargs.version()` functionality when yargs is linked:

`npm link`
`npm link yargs`

2. when testing I sometimes like to create a bin in the root directory of the yargs folder, this would also not work for `yargs.version()`.

My temptation is to handle the edge-case, and add tests for any weird exceptions we run into in the wild.

what do you think?